addons/account/models/account_account.py

```diff
@@ -858,6 +1089,262 @@ def get_import_templates(self):
+            query = SQL(
+                """
+                UPDATE %(table)s
+                SET %(column)s = %(account_id_to_merge_into)s
+                WHERE %(column)s IN %(account_ids_to_remove)s
+                    AND %(model_column)s = 'account.account'
+                """,
+                table=SQL.identifier(self.env[model]._table),
+                column=SQL.identifier(field_to_update.name),
+                account_id_to_merge_into=account_to_merge_into.id,
+                account_ids_to_remove=tuple(accounts_to_remove.ids),
+                model_column=model_column,
+            )
+            operations[model].append((field_to_update, records_to_update, query))
+
+        # 4.5: Execute UPDATEs
+        models_to_ignore_security_checks = [
+            'account.invoice.report',  # Everything is done in sudo in this model, but we shouldn't block merging accounts.
+        ]
+        for model, model_operations in operations.items():
+            if model not in models_to_ignore_security_checks:
+                # Check access rights
+                self.env[model].check_access_rights('write')
+                self.env[model].check_field_access_rights('write', [f for f, _, _ in model_operations])
+
+            for field_to_update, records_to_update, query in model_operations:
+                if model not in models_to_ignore_security_checks:
+                    # Check record rules
+                    records_to_update.check_access_rule('write')
+
+                records_to_update.invalidate_recordset()
+                self.env.cr.execute(query)
+
+        # 4.6: Update ir_property. First, don't update values if they would trigger the unique constraint.
+        self.env['ir.property'].invalidate_model()
+        self.env.cr.execute(SQL(
+            """
+            UPDATE ir_property AS _ip1
+            SET res_id = %(updated_value)s
+            WHERE res_id IN %(values_to_update)s
+            AND NOT EXISTS (
+                SELECT
+                FROM ir_property AS _ip2
+                WHERE _ip2.res_id = %(updated_value)s
+                AND _ip2.fields_id = _ip1.fields_id
+                AND _ip2.company_id = _ip1.company_id
+            )""",
+            updated_value=f'account.account,{account_to_merge_into.id}',
+            values_to_update=tuple(f'account.account,{account_id}' for account_id in accounts_to_remove.ids),
+        ))
+
+        # Then delete any entries that were not updated by the query above
+        self.env.cr.execute(SQL(
+            """
+            DELETE FROM ir_property
+            WHERE res_id IN %(values_to_delete)s
+            """,
+            values_to_delete=tuple(f'account.account,{account_id}' for account_id in accounts_to_remove.ids),
+        ))
+
+        # Step 5: Remove merged accounts
+        accounts_to_remove.unlink()
```

Comment:
Might as well delete in SQL to avoid checking that the account is not used anymore? (and other similar potential constraints)